### PR TITLE
Alonfaraj fixes

### PR DESCRIPTION
**Pull request by alonfaraj**

### CODE REVIEW

File: OgreMain/src/OgreScriptTranslator.cpp [hunk 1]
```diff
@@ -3841,7 +3841,7 @@ namespace Ogre{
         // Set the custom parameters
         for(const auto& p : customParameters)
         {
-            if(prog->isSupported() && !prog->setParameter(p.first->name, p.second))
```
Comment (paroj) [inline OgreMain/src/OgreScriptTranslator.cpp]: we need this so parameters of unsupported programs will not generate false positives. Think of having GLSL and HLSL shaders in of material file.

Also `mSyntaxCode` is initialised by all programs to sane values on construction. If it is `""` for you, the error must be somewhere else.

Comment (alonfaraj) [reply OgreMain/src/OgreScriptTranslator.cpp]: ok, thanks, I will check it again.

The second commit is not related to this one, I will wait for your review.

Comment (paroj) [reply OgreMain/src/OgreScriptTranslator.cpp]: the second commit is fine, however there is no easy way to merge individual commits of a PR.

Comment (alonfaraj) [reply OgreMain/src/OgreScriptTranslator.cpp]: my bad, it's the my first PR, sorry for that.
I will make a new one.

Comment (paroj) [reply OgreMain/src/OgreScriptTranslator.cpp]: if you know your way around git, it is enough to force-push to the existing branch, without creating a new PR.

**Comment by paroj** [comment]
closing this for now

**Pull request closed without merging**